execution.py

```diff
@@ -34,12 +39,39 @@ def get_input_data(inputs, class_def, outputs={}, prompt={}, extra_data={}):
                     input_data_all[x] = extra_data['extra_pnginfo']
     return input_data_all
 
+def is_node_mode(mode, prompt, unique_id, follow_connections=False):
```

Comment:
Instead of doing this why not just not include the muted node the prompt json sent to the backend?

It should have the same effect.

Reply:
Let's take the workflow above as example, when the LatentUpscale is muted it should prevents not just the LatentUpscale from executing but also the dependents nodes KSampler, VAEDecodeTiled and SaveImage.

Without it the KSampler will execute and expects to have the latent_image input but it won't have it because the LatentUpscale won't run because it is muted.

The prompt json sent to the backend just includes the LatentUpscale as muted, but for that to work it needs to mute all dependents nodes, I opted to spread the mute in backend to make it ease for api calls to mute nodes, the client just needs to mute one node and all the subsequent nodes are considered as muted by the backend.

Reply:
Isn't that exactly what happens right now if you just delete the node?

Reply:
> Isn't that exactly what happens right now if you just delete the node?

When delete the node yes, but when the node is present and muted it get a error.

But I found a better way of doing it.

Reply:
Wait, I need to commit other things.

Reply:
Is there a reason the backend needs to know specifically that the node is muted? Wouldn't just skipping the serialization of the muted nodes be simpler?

Reply:
> Is there a reason the backend needs to know specifically that the node is muted? Wouldn't just skipping the serialization of the muted nodes be simpler?

This is so that the entire prompt is saved, so you can generate the image with muted optional processes and share the image, who gets the image can activate those optional processes, that can be anything, like upscale, different styles, that way you can share multiple alternatives in one image, without having to use everything at the same time.